src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs

```diff
@@ -133,8 +133,16 @@ private bool ShouldSerialize(OptionKey optionKey)
 
         public override OptionSet WithChangedOption(OptionKey optionKey, object? value)
         {
-            // make sure we first load this in current optionset
-            this.GetOption(optionKey);
+            // Make sure we first load this in current optionset
+            var currentValue = this.GetOption(optionKey);
+
+            // Check if the new value is the same as the current value.
+            if (value == currentValue)
```

Comment:
💭 Should this use the same equality logic that `WithLanguages` uses?

https://github.com/dotnet/roslyn/blob/3d333c226822a88df0e1a585f56f498299797c7f/src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs#L110

Reply:
Yep, done.